x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/DeploymentManager.java

```diff
@@ -84,26 +89,56 @@ public void stopDeployment(TrainedModelDeploymentTask task) {
         }
         if (processContext != null) {
             logger.debug("[{}] Stopping deployment", task.getModelId());
-            processContext.killProcess();
+            processContext.stopProcess();
         } else {
             logger.debug("[{}] No process context to stop", task.getModelId());
         }
     }
 
+    public void infer(TrainedModelDeploymentTask task, double[] inputs, ActionListener<PyTorchResult> listener) {
+        ProcessContext processContext = processContextByAllocation.get(task.getAllocationId());
+        try {
+            String requestId = processContext.process.get().writeInferenceRequest(inputs);
+            waitForResult(processContext, requestId, listener);
```

Comment:
A future improvement would be to not block the thread here

Reply:
I think this might be worth fixing now. It's something that could be forgotten with terrible consequences. I'll work on this following the pattern we used in `AutodetectCommunicator`.

Reply:
Actually, the pattern in `AutodetectCommunicator` ensures we only perform one operation at a time against the native process. This might not be a restriction for pytorch. We could investigate multithreading capabilities in order to do inference on multiple requests in parallel in the same process. So, for now, I will just make sure we're not blocking the thread.